dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/validation/AppValidator.java

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.reactive.messaging.fat.validation;
+
+import static com.ibm.ws.microprofile.reactive.messaging.fat.kafka.common.KafkaUtils.kafkaClientLibs;
+import static com.ibm.ws.microprofile.reactive.messaging.fat.kafka.common.KafkaUtils.kafkaPermissions;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+
+import com.ibm.websphere.simplicity.PropertiesAsset;
+import com.ibm.websphere.simplicity.RemoteFile;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.matchers.Matchers;
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * Helper class for building and testing apps that should fail to start in a particular way
+ * <p>
+ * Example:
+ *
+ * <pre>
+ * {@code
+ * AppValidator.validateAppOn(SHARED_SERVER)
+ *                 .withClass(MyInvalidClass.class)
+ *                 .failsWith("CWMFT5001E")
+ *                 .run();
+ * }
+ * </pre>
+ *
+ * <p>
+ * This creates a web application containing just {@code MyInvalidClass}, deploys it, checks that it fails to start and that a message containing {@code CWMFT5001E} is found in the
+ * log and then undeploys it.
+ */
+public class AppValidator {
+    private final LibertyServer server;
+    private final List<Class<?>> classes;
+    private final List<String> stringsToFind;
+    private PropertiesAsset appConfig;
+
+    private static String APP_START_CODE = "CWWKZ000[13]I";
+    private static String APP_FAIL_CODE = "CWWKZ000[24]E";
+    private static int appCount = 0;
+
+    /**
+     * Begin setting up an app validation to run on the given liberty server
+     */
+    public static AppValidator validateAppOn(LibertyServer server) {
```

Comment:
One frustration I have with this class is that we effectively end up with an arbitary testApp name which can only be matched by a regex as you have done, but there is no way to guarantee within a test what that app name is and we are checking that in the logs. I would recommend added a property called appName that can be set on the AppValidator before `.run()` is called and uses that. you can then set them in the individual tests and then check for that explicit appName and that will reduce potential false-postivites. It should use the mechanism already used here if appName is null